pallets/multicollateral-bonding-curve-pool/src/migrations/v4.rs

```diff
@@ -0,0 +1,103 @@
+use crate::Config;
+use crate::Pallet;
+use common::AssetIdOf;
+use common::Balance;
+use frame_support::log::{error, info};
+use frame_support::pallet_prelude::*;
+use frame_support::traits::Get;
+use frame_support::traits::OnRuntimeUpgrade;
+use sp_runtime::traits::One;
+use sp_std::collections::btree_map::BTreeMap;
+
+#[cfg(feature = "try-runtime")]
+use sp_std::prelude::Vec;
+
+pub mod old_storage {
+    use super::*;
+    #[frame_support::storage_alias]
+    pub type PendingFreeReserves<T: Config> =
+        StorageValue<Pallet<T>, Vec<(AssetIdOf<T>, Balance)>, ValueQuery>;
+}
+
+pub struct MigrateToV4<T>(core::marker::PhantomData<T>);
+
+impl<T> OnRuntimeUpgrade for MigrateToV4<T>
+where
+    T: crate::Config,
+{
+    fn on_runtime_upgrade() -> frame_support::weights::Weight {
+        if Pallet::<T>::on_chain_storage_version() != 3 {
+            error!(
+                "Runtime upgrade executed with wrong storage version, expected 3, got {:?}",
+                Pallet::<T>::on_chain_storage_version()
+            );
+            return <T as frame_system::Config>::DbWeight::get().reads(1);
+        }
+        info!("Applying migration to version 4: Move free reserves distribution to Hooks");
+        let pending_free_reserves = old_storage::PendingFreeReserves::<T>::take()
+            .into_iter()
+            .collect::<BTreeMap<AssetIdOf<T>, Balance>>();
+        if !pending_free_reserves.is_empty() {
+            info!("Migrate pending free reserves: {:?}", pending_free_reserves);
+            crate::PendingFreeReserves::<T>::insert(
+                frame_system::Pallet::<T>::block_number() + One::one(),
```

Comment:
```suggestion
                frame_system::Pallet::<T>::block_number() + T::BlockNumber::one(),
```
Bit more explanatory.